### ISSUE TITLE
Infinite loop

### ISSUE THREAD

**Issue by tombentley** [rohitmohan96/ceylon.markdown#10]
Slightly malformed markdown can cause an infinite loop:

```
shared void bug() {
    parse("""a paragraph
                 code block without a blank line*

             whatevs.""");
}
```

**Comment by rohitmohan96** [comment]
Thanks. This should be fixed in 70e42a1c6a8572ad030f26ba29a0b8a53d4321de. This also fixed some of my JS tests that did not work.

**Comment by rohitmohan96** [comment]
Closing this for now.
